### DISCUSSION TITLE
sonoff Basic/PowR2/... : delay when using physical button

### DISCUSSION THREAD

**Discussion by bartplessers** [arendst/Tasmota#13974] [Q&A] [answered]
Hi,

I tasmotized some Sonoff devices.
Now I notice that when I press the (physical) button on the device, the relay switches on with some delay (or on "button release"? have to figure out this this evening).
If you use the native firmware, the relay switches immediately (or on "button press"?)

Not a big deal, because most of the time the devices are controlled by mqtt and/or Tasmota integration in hass.
But I'm curious: can the behavior altered? So that pressing button will immediately turn on a device? How?

thanx in advance,
Bart

**Comment by sfromis** [comment, marked as answer]
What is "some delay"? Is it more than simply button release? If the latter, [SetOption13](https://tasmota.github.io/docs/Commands/#setoption13) allows immediate action on single button press.
